### DISCUSSION TITLE
SSH key management

### DISCUSSION THREAD

**Discussion by teintuc** [Dashlane/dashlane-cli#199] [Ideas]
Hello all,

First thanks for the cli. I have been waiting for it for a while !

I made a shell script that read the vault in order to retrieve a ssh private-key/passphrase and inject them into the running ssh agent. It works and it is nice for my use.

Now, I was thinking to go further. 
May be we can add those capabilities directly to the cli ? Like `dcli ssh [title]` and it will add that key into the ssh-agent (ask the passphrase if there is none setup in dashlane vault).

Other idea: Make a custom ssh-agent, like a dashlane ssh-agent. One that read in the vault first and then go back into a normal use. 
It is a bigger project but we wouldn't need to add the keys in the agent before.

I'm trying to POC it but I'm not a TS dev. More Golang :)
Do you know any ssh TS libraries that could help me ?

Thanks

**Comment by michaelkhalsa** [comment]
I second that as a nice feature to add.
